repository/kafka/docs/latest/production.md

```diff
@@ -0,0 +1,169 @@
+pvc-de527673-8bee-4e38-9e6d-399ec07c2728   5Gi        RWO            Delete           Bound    default/kafka-datadir-kafka-kafka-1                                awsebscsiprovisioner            18h
+```
+
+We can patch the `PersistentVolumes` to use the `ReclaimPolicy` of `Retain`
+
+```
+kubectl patch pv pvc-6a9e69f4-b807-440f-b190-357c109e8ad9 -p '{"spec":{"persistentVolumeReclaimPolicy":"Retain"}}'
+kubectl patch pv pvc-8602a698-14a0-4c3d-85e6-67eb6da80a5d -p '{"spec":{"persistentVolumeReclaimPolicy":"Retain"}}'
+kubectl patch pv pvc-de527673-8bee-4e38-9e6d-399ec07c2728 -p '{"spec":{"persistentVolumeReclaimPolicy":"Retain"}}'
+
+persistentvolume/pvc-6a9e69f4-b807-440f-b190-357c109e8ad9 patched
+persistentvolume/pvc-8602a698-14a0-4c3d-85e6-67eb6da80a5d patched
+persistentvolume/pvc-de527673-8bee-4e38-9e6d-399ec07c2728 patched
+```
+
+Verify that `ReclaimPolicy` has been changed for the broker `PersistentVolumes`
+
+```
+> kubectl get pv
+pvc-6a9e69f4-b807-440f-b190-357c109e8ad9   5Gi        RWO            Retain           Bound    default/kafka-datadir-kafka-kafka-0                                awsebscsiprovisioner            18h
+pvc-8602a698-14a0-4c3d-85e6-67eb6da80a5d   5Gi        RWO            Retain           Bound    default/kafka-datadir-kafka-kafka-2                                awsebscsiprovisioner            18h
+pvc-de527673-8bee-4e38-9e6d-399ec07c2728   5Gi        RWO            Retain           Bound    default/kafka-datadir-kafka-kafka-1                                awsebscsiprovisioner            18h
+```
+
+
+
+## Broker configuration
+
+#### Replication Factor
+
+Change the default replication factor from the default 1 to 3
+
+```
+DEFAULT_REPLICATION_FACTOR=3
+```
+
+#### Minimum Insync Replicas
+
+When a producer sets acks to `all` (or `-1`) `MIN_INSYNC_REPLICAS` is the minimum number of replicas that must acknowledge a write for the write to be considered successful.
+When used together, `min.insync.replicas` and `acks` allow Apache Kafka users to enforce greater durability guarantees.
+
+To guarantee the messages durability a recommended practice would be to create a topic with a replication factor of `3`, set `MIN_INSYNC_REPLICAS` to `2`, and produce messages with acks of `all`.
+
+```
+MIN_INSYNC_REPLICAS=2
+```
+
+#### Number of partitions
+
+The number of partitions should be higher than `1` for reliability reasons, by default this option is set to `3`
+
+More partitions lead to higher throughput performance but affect latency and availability. In KUDO Kafka we can set this trade-off of throughput vs. availability by setting the default partition number.
+
+```
+NUM_PARTITIONS=3
+```
+
+#### Graceful rolling restarts
+
+KUDO Kafka enables by default the option `CONTROLLED_SHUTDOWN_ENABLE` to true. To ensure a faster flush during shutdown and faster recovery during the bootstrap `num.recovery.threads.per.data.dir` can be increased from default `1` thread.
```

Comment:
```suggestion
By default, KUDO Kafka sets `CONTROLLED_SHUTDOWN_ENABLE` to `true`. To ensure a faster flush during shutdown, and faster recovery during bootstrap, `num.recovery.threads.per.data.dir` can be increased from the default of `1` thread.
```

I'm a little lost on how `CONTROLLED_SHUTDOWN_ENABLE` came into this picture when discussing recovery.threads. Perhaps provide a small explanation on controlled_shutdown first?